server/src/services/commits.ts

```diff
@@ -94,11 +94,23 @@ const addCommit = async (
     );
   }
 
-  return await commitStorage.addCommit({
+  return commitStorage.addCommit({
     ...DEFAULT_COMMIT_PAYLOAD,
     ...commitData,
     createdAt: new Date(),
   });
 };
 
-export default {getAllCommits, addCommit};
+/**
+ * Deletes the commit with the specified commitId.
+ * An error would be thrown if commit does not already exist, and if deletion fails.
+ * @param commitId Id of the commit to be deleted
+ */
+const deleteCommit = async (commitId: number) => {
+  const commit = await commitStorage.getCommit(commitId);
```

Comment:
You should have check on commit status right? e.g. can you delete a paid commit?

Reply:
@ifhuang Yes I should check if it's ongoing! Oops was gonna do that but I got distracted by the datastore method and it slipped my mind, sorry! 😅